monty.qmd

```diff
@@ -1,16 +1,95 @@
+
+## A simple example
+
+```{r}
+library(monty)
+```
+
+We can define a simple Gaussian [mixture model](https://en.wikipedia.org/wiki/Mixture_model) of two sub-populations using the `monty_model_function()` from the package. The model represents the distribution of a quantity $l$, sampled with a probability $p$ from a normal distribution of mean $m_{1}$ and with a probability $1-p$ from another normal distribution of mean $m_{2}$. Both subpopulations have variance equal to 1. 
+
+To build our `monty` model, we start by defining an R function returning the log-density of our statistical model. This function has for arguments $l$, the quantity of interest and the three parameters $p$, $m_{1}$ and $m_{1}$ defining the two Normally-distributed subpopulations.
+
+```{r}
+fn <- function(l, p, m1, m2) {
+  log(p*dnorm(l,mean = m1) + (1-p)*dnorm(l, mean = m2))
+}
+```
+
+Assuming that the population is 75% from subpopulation 1 (normally distributed with mean 3) and 25% from subpopulation 2 (also normally distributed but with mean 7), we can build our `monty` model by indicating that the parameters of the subpopulations are fixed at these given values.
+
+```{r}
+l_distribution <- monty_model_function(fn,
+                                       fixed = list(p = 0.75, m1 = 3, m2=7))
+```
+
+We have just created a `monty` model. 
+
+```{r}
+l_distribution
+```
+
+We can plot the density of our model for values of $l$ between 0 and 10 and eye-check that we can see the two subpopulations in the correct proportions and with the 3 and 7 as modes.
+
+```{r}
+l <- seq(from = 0, to = 10, by = 0.1)
+plot(l,
+     exp(Vectorize(l_distribution$density)(l)),
+     type = "l",
+     ylab = "density")
+```
+
+## Sampling from our example distribution
+
+We want now to sample from this model, using the `monty_sample()` function. For this we need to tell `monty` which sampler we want to use to explore our distribution. There are a variety of samplers avalaible and you can learn about them in @sec-samplers. One of the simplest is the random walk [Metropolis-Hastings](https://en.wikipedia.org/wiki/Metropolis%E2%80%93Hastings_algorithm) algorithm that should work almost out of the box (though not necesseraly efficiently) in most cases.
+
+The random walk sampler uses a variance-covariance (VCV) matrix to guide its exploration, determining the 'jump' from the current point to the next in a random walk by drawing from a multivariate normal distribution parameterised by this matrix. For a single-parameter model, we use a 1x1 matrix (`matrix(1)`) as our VCV matrix.
+
+The choice of the VCV matrix is critical for the sampler's efficiency, especially in more complex cases where the tuning of this matrix can significantly affect performance. A well-chosen VCV matrix optimises moving accross the parameter space, making the random walk sampler more effective in exploring the distribution of interest.
+
+```{r}
+sampler <- monty_sampler_random_walk(matrix(1))
```

Comment:
People may get confused with this being `matrix(1)` (it's a 1x1 matrix with a value of 1). If you used `matrix(0.5)` (say) then it would be easier to talk about (a 1x1 matrix with a variance in the proposal of 0.5)

Reply:
Done. Changed to a variance of 2 - not much of a difference in sampling performance, with proposal steps on average sqrt(2) bigger.